### ISSUE TITLE
Failed to parse error output when not authorised

### ISSUE THREAD

**Issue by martinda** [cdancy/bitbucket-rest#140]
### Expected Behavior

When calling `pullRequestApi().get()` without proper authorisation to the REST endpoint, an  error should be reported in the list of `errors()` of the pull-request instance, just like when the same operation is attempted with curl:

```
$ curl -vv -X GET "http://localhost:7990/rest/api/1.0/projects/PROJECT/repos/REPO/pull-requests/1"
...
{"errors":[{"context":null,"message":"You are not permitted to access this resource","exceptionName":"com.atlassian.bitbucket.AuthorisationException"}]}
```

### Current Behavior

The error cannot be parsed, and is instead returned as:
```
Error{context={"context":null,"message":"You are not permitted to access this resource","exceptionName":null}, message=Failed to parse output: message=null, exceptionName=java.lang.NullPointerException, conflicted=false, vetoes=[]}
```
In this case, the `error()` object contains a message that says that the message is null. This hides the real error.

I tried to trace the root of the problem in the source and I am baffled. It seems that inside `BitbucketFallbacks.java` the `getErrors(final String output)` method argument is not getting the correct value of the `output` input parameter. It gets this:

```
{"context":null,"message":"You are not permitted to access this resource","exceptionName":null}
```
It is like the "errors" array is stripped out by the `org.jclouds.rest.AuthorizationException`. I would expect to see the same as what curl is reporting. Consequently, the line that tries to get the "errors" Json array throws an exception, and the real error message is not in the message element.

### Context

I am testing for invalid user inputs.

### Steps to Reproduce (for bugs)

This short groovy script can reproduce the problem:
```groovy
@Grab(group='com.cdancy', module='bitbucket-rest', version='2.1.2')

import com.cdancy.bitbucket.rest.BitbucketClient
import com.cdancy.bitbucket.rest.domain.pullrequest.PullRequest

BitbucketClient client = BitbucketClient.builder().build()

PullRequest pr = client.api().pullRequestApi().get("abc", "def", 1)
pr.errors().each { error ->
    println(error)
}
```
### Your Environment

Linux command line.

**Comment by cdancy** [comment]
I suspect that's where the issue is. I'll have a look at things today (assuming you don't beat me to it).

**Comment by martinda** [comment]
Looks like not all installations of bitbucket return the same json response. For example, on my home installation, I get this:
```bash
$ curl -vv "http://localhost:9990/rest/api/1.0/projects/PROJECT/repos/REPO/pull-requests/1"
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 9990 (#0)
> GET /rest/api/1.0/projects/PROJECT/repos/REPO/pull-requests/1 HTTP/1.1
> Host: localhost:9990
> User-Agent: curl/7.47.0
> Accept: */*
> 
< HTTP/1.1 401 
< X-AREQUESTID: @HDBYKNx1281x125x0
< X-ASEN: SEN-L11638810
< Cache-Control: no-cache, no-transform
< Vary: X-AUSERNAME,Accept-Encoding
< Transfer-Encoding: chunked
< Content-Type: application/json;charset=UTF-8
< X-Content-Type-Options: nosniff
< WWW-Authenticate: OAuth realm="http%3A%2F%2Flocalhost%3A9990"
< Date: Wed, 02 May 2018 01:21:48 GMT
< 
* Connection #0 to host localhost left intact
{"context":null,"message":"You are not permitted to access this resource","exceptionName":null}
```
And for another REST point I get the [documented](https://docs.atlassian.com/bitbucket-server/rest/5.10.0/bitbucket-rest.html#errors-and-validation) json with the context inside a list of `errors`:
```bash
$ curl -vv "http://localhost:9990/rest/api/1.0/users"
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 9990 (#0)
> GET /rest/api/1.0/users HTTP/1.1
> Host: localhost:9990
> User-Agent: curl/7.47.0
> Accept: */*
> 
< HTTP/1.1 401 
< X-AREQUESTID: @HDBYKNx1281x124x0
< X-ASEN: SEN-L11638810
< Cache-Control: no-cache, no-transform
< Vary: X-AUSERNAME,Accept-Encoding
< Transfer-Encoding: chunked
< Content-Type: application/json;charset=UTF-8
< X-Content-Type-Options: nosniff
< WWW-Authenticate: OAuth realm="http%3A%2F%2Flocalhost%3A9990"
< Date: Wed, 02 May 2018 01:21:39 GMT
< 
* Connection #0 to host localhost left intact
{"errors":[{"context":null,"message":"You are not permitted to access this resource","exceptionName":"com.atlassian.bitbucket.AuthorisationException"}]}
```
Maybe the fallback handler needs to become more flexible in what it can take.

**Comment by martinda** [comment]
The `getError()` method in `BitbucketFallbacks.java` is completely unaware of the HTTP status, and it relies only on the json provided by Bitbucket for creating the error message. On the other hand the [jenkins-rest](https://github.com/cdancy/jenkins-rest), fallbacks are aware of the HTTP status. What do you think should be the right solution here?

**Comment by cdancy** [comment]
Yeah what I did with `jenkins-rest` was "more correct" after having not looked at this code in a while.

The returned response being different is indeed odd and something we'll have to account for.

**Comment by martinda** [comment]
The nice thing with the different approaches between jenkins-rest and bitbucket-rest you get to see the pros and cons of both (and so do I). Learning is always good.

I made changes to `getError()` for now with a few new tests to cover it, and all tests are passing. I will continue to use my snapshot today, and propose my changes as a PR in a few days. Gives time to think.

**Comment by cdancy** [comment]
@martinda sounds good and thanks! I'm actually traveling at the moment and won't be able to really look at things until at least tomorrow. As it seems you have something going I'll hold off and wait for your PR and we can build on that.

**Comment by cdancy** [comment]
Possible fix put together at #141
